node/src/components/block_validator.rs

```diff
@@ -46,6 +50,11 @@ pub enum Event<T, I> {
     /// A request to find a specific deploy, potentially from a peer, failed.
     #[display(fmt = "deploy {} missing", _0)]
     DeployMissing(DeployHash),
+
+    /// An event changing the current state to BlockValidatorReady, once the chainspec has been
+    /// loaded.
+    #[display(fmt = "block validator loaded")]
+    Loaded { chainspec: Arc<Chainspec> },
```

Comment:
We could also use the approach that @marc-casperlabs used in the `BlockProposer` – basically turning it into an actor-like component that loads its state prior to launch.

Reply:
I largely followed what had been done in the `block_proposer` - what improvements do you think could be made, or did I miss something?